core/trino-main/src/main/java/io/trino/operator/aggregation/AccumulatorCompiler.java

```diff
@@ -134,14 +133,26 @@ public static GenericAccumulatorFactoryBinder generateAccumulatorFactoryBinder(A
             lambdaProviderFields.add(definition.declareField(a(PRIVATE, FINAL), "lambdaProvider_" + i, LambdaProvider.class));
         }
 
-        FieldDefinition inputChannelsField = definition.declareField(a(PRIVATE, FINAL), "inputChannels", type(List.class, Integer.class));
-        FieldDefinition maskChannelField = definition.declareField(a(PRIVATE, FINAL), "maskChannel", type(Optional.class, Integer.class));
+        FieldDefinition maskChannelField = definition.declareField(a(PRIVATE, FINAL), "maskChannel", int.class);
+
+        int inputChannelCount = countInputChannels(metadata.getValueInputMetadata());
+        List<FieldDefinition> inputChannelFields;
+        if (inputChannelCount == 0) {
+            inputChannelFields = ImmutableList.of();
+        }
```

Comment:
No need to special case `== 0`. The result will be the same as going through the loop below.

Reply:
Removed